### PR TITLE
Update README.md (repo structure, features)

### PR DESCRIPTION
**Pull request by tedasdf**
* Update README.md

added title/project description

* Update README.md

* Update README.md

* Update README.md

* Update README.md

---------

### 🤔 Reasoning

_Explain the purpose of this PR..._

### 🚧 Changes

_Describe the changes made..._

### ✅ PR Checklist

- [x] Using this PR template?
- [x] Linked issue?
- [ ] Added feature?
    - [ ] Added/updated docs?
    - [ ] Added/updated tests?

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,8 +1,103 @@
-# discussion-agents
 
-Discussion agents. 
 
-# Developer Setup: Windows
+
+
+# Discusion Agents
```
Comment (alckasoc) [inline README.md:5]:
```suggestion
# Discussion Agents
```

File: README.md [hunk 2]
```diff
@@ -1,56 +1,103 @@
...
-## ReAct 
+- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.
 
+- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.
+
+- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.
+
+- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.
+
+# The library offers:
```
Comment (alckasoc) [inline README.md:32]: Merge features, and the library offers.

File: README.md [hunk 3]
```diff
@@ -1,56 +1,103 @@
...
 
+#### Getting started example
 
+```python
+question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring'
 
-SYNERGIZING REASONING AND ACTING IN LANGUAGE MODELS (https://arxiv.org/pdf/2210.03629.pdf)
 
+llm = OpenAI(openai_api_key="YOUR_API_KEY")
+agent = ReActAgent(llm=llm)
```
Comment (alckasoc) [inline README.md:47]: not `OpenAI`. The `llm` parameter takes a `BaseChatModel`.

File: README.md [hunk 4]
```diff
@@ -1,56 +1,103 @@
...
+out = agent.generate(question=question)
 
-## Reflexion 
+```
 
 
-Concept
+Output
```
Comment (alckasoc) [inline README.md:53]: Shouldn't include full output.

File: README.md [hunk 5]
```diff
@@ -1,56 +1,103 @@
...
+
+**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about the controversies and crimes.
+
+**Action**: Lookup[unsportsmanlike conducts]
+
+...
+
+---
+
+The iterative process of thought, action, and observation offers an intuitive and transparent approach to problem-solving.
```
Comment (alckasoc) [inline README.md:85]: delete

File: README.md [hunk 6]
```diff
@@ -1,56 +1,103 @@
...
+
+
+
+
+
+
+
+
+
+## Developer Setup: Windows
```
Comment (alckasoc) [inline README.md:100]:
```suggestion
## Developer Setup
```

**Review by alckasoc — CHANGES_REQUESTED** [review]

File: README.md (hunk 4 above)
Comment (tedasdf) [reply README.md:53]: its not the full output

Comment (tedasdf) [reply README.md:53]: should i shorten the output more

Comment (alckasoc) [reply README.md:46]: yep

File: README.md [hunk 7]
```diff
@@ -1,8 +1,69 @@
...
+
+## Features
+
+
+Welcome to our machine learning library leveraging the power of LLM (Large Language Model). Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.
+
+
+
+
+- LLM Integration: Leverages the advanced language processing capabilities of LLM for enhanced accuracy and efficiency.
```
Comment (alckasoc) [inline README.md:13]:
```suggestion
```

File: README.md [hunk 8]
```diff
@@ -1,8 +1,69 @@
...
+
+- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.
+
+- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.
+
+## The library offers:
+
+
+- Modularized implementations of popular LLM-based agents and agent-related methods.
+- A flexible framework to build your own agents using innovations from the literature.
```
Comment (alckasoc) [inline README.md:27]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -1,8 +1,69 @@
...
+
+
+
+#### Getting started example
+
+```python
+question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring'
```
Comment (alckasoc) [inline README.md:34]:
```suggestion
question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'
```

File: README.md [hunk 10]
```diff
@@ -1,8 +1,69 @@
...
+
+```
+
+
+Output
+----
+
+**Question** :
+
+I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
```
Comment (alckasoc) [inline README.md:49]: incorrect formatting

File: README.md [hunk 11]
```diff
@@ -31,45 +92,71 @@ Finally install all of the packages.
...
-    │   ├── agent            <- Model/agent-related modules.
-    │   │   └── ...
-    │   │
-    │   ├── core             <- Agent core modules.
-    │   │   └── ...
+    ├── discussion_agents                   <- Source code for this project.
+    │   ├── cog   
+    │   │   ├── agent                       <- Model/agent-related modules.
+    │   │   │   ├── base.py                 <-
+    │   │   │   ├── generative_agents.py
```
Comment (alckasoc) [inline README.md:116]: don't need to specify every single file within a module. not scalable

**Review by alckasoc — CHANGES_REQUESTED** [review]

File: README.md (hunk 10 above)
Comment (tedasdf) [reply README.md:49]: so do you want me to do it as same as the Acts and Observataions

File: README.md (hunk 11 above)
Comment (tedasdf) [reply README.md:116]: then i will just have the folders

File: README.md [hunk 12]
```diff
@@ -1,8 +1,57 @@
...
+
+```
+
+
+Output
+----
+
+**Question** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
+
+
```
Comment (alckasoc) [inline README.md:39]: missing a thought?

File: README.md [hunk 13]
```diff
@@ -31,45 +80,51 @@ Finally install all of the packages.
...
+    │   │   │   
+    │   │   ├── eval                        <- Agent core modules.
+    │   │   │   
+    │   │   ├── functional                  
+    │   │   │
+    │   │   ├── modules           
+    │   │   │   ├── memory      <- Memory-related modules.
+    │   │   │   ├── plan        <- Planning-related modules.
+    │   │   │   ├── reflect      <- Reflecting-related modules.
+    │   │   │   └── score        <- Scoring-related modules.
```
Comment (alckasoc) [inline README.md:112]:
```suggestion
    │   │   │   ├── memory      <- Memory-related modules.
    │   │   │   ├── plan            <- Planning-related modules.
    │   │   │   ├── reflect         <- Reflecting-related modules.
    │   │   │   └── score          <- Scoring-related modules.
```

File: README.md [hunk 14]
```diff
@@ -31,45 +80,51 @@ Finally install all of the packages.
...
-    │   │   └── ...
-    │   │
-    │   ├── scoring          <- Scoring-related modules.
-    │   │   └── ...
-    │   │
-    │   ├── utils            <- Utility methods.
-    │   │   └── ...
+    │   └── utils           <- Utility methods.
     │
     ├── docs                 <- A default Sphinx project; see sphinx-doc.org for details.
```
Comment (alckasoc) [inline README.md:120]:
```suggestion
    │   └── utils             <- Utility methods.
    │
    ├── docs                 <- A default Sphinx project; see sphinx-doc.org for details.
```

**Review by alckasoc — CHANGES_REQUESTED** [review]

File: README.md [hunk 15]
```diff
@@ -1,8 +1,57 @@
...
+
+**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
+
+**Action**: Lookup[crimes]
+
+**Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
+
+**Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
+
+...
```
Comment (alckasoc) [inline README.md:52]:
```suggestion
**Question** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.

**Action**: Search[best kick boxer in the world]
**Observation**: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
**Action**: Lookup[crimes]
**Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
**Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
**...**
```

File: README.md [hunk 16]
```diff
@@ -31,45 +80,51 @@ Finally install all of the packages.
...
 ------------
 
+ 
+
     ├── data
-    │   ├── external         <- Data from third party sources.
-    │   ├── interim          <- Intermediate data that has been transformed.
-    │   ├── processed        <- The final, canonical data sets for modeling.
-    │   └── raw              <- The original, immutable data dump.
+    │   ├── external                        <- Data from third party sources.
```
Comment (alckasoc) [inline README.md:95]: Need to make sure all arrows line up

**Review by alckasoc — CHANGES_REQUESTED** [review]

File: README.md [hunk 17]
```diff
@@ -1,8 +1,52 @@
...
+- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.
+
+- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.
+
+- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.
+
+- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.
+
+
+#### Getting started example
```
Comment (alckasoc) [inline README.md:21]: ## Getting Started

File: README.md [hunk 18]
```diff
@@ -1,8 +1,52 @@
...
+
+```python
+question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'
+
+
```
Comment (alckasoc) [inline README.md:26]:
```suggestion

```

File: README.md [hunk 19]
```diff
@@ -1,8 +1,52 @@
...
+**Thought** : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
+
+**Action**: Search[best kick boxer in the world]
+**Observation**: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
+**Thought**: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
+**Action**: Lookup[crimes]
+**Observation**: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
+**Thought**: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
+**...**
```
Comment (alckasoc) [inline README.md]:
```
Thought: I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
Action: Search[best kick boxer in the world]
Observation: (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
Thought: It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
Action: Lookup[crimes]
Observation: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
Thought: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
...
```

File: README.md [hunk 20]
```diff
@@ -1,8 +1,55 @@
...
+
+ Our primary goal is to provide easy-to-use and clean functions for natural language processing tasks, enabling developers to build robust and efficient applications with ease.
+
+
+
+- Easy-to-Use Interface: Provides intuitive and user-friendly functions for rapid prototyping and development.
+
+- Clean Functions: Offers clean and well-structured functions, promoting readability and maintainability of code.
+
+- Scalability: Designed for scalability, enabling processing of large datasets and complex models with ease.
```
Comment (alckasoc) [inline README.md:16]:
```suggestion
```

File: README.md [hunk 21]
```diff
@@ -1,8 +1,55 @@
...
+
+- Modularized Implementations: Includes modularized implementations of popular LLM-based agents and agent-related methods, allowing users to leverage cutting-edge innovations from the literature.
+
+
+#### Getting started
```
Comment (alckasoc) [inline README.md:21]:
```suggestion
## Getting Started 
```

File: README.md [hunk 22]
```diff
@@ -1,8 +1,55 @@
...
+question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'
+
+llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
+agent = ReActAgent(llm=llm)
+out = agent.generate(question=question)
+
+```
+
+
+Output
```
Comment (alckasoc) [inline README.md:33]:
```suggestion
```

**Review by alckasoc — CHANGES_REQUESTED** [review]

File: README.md [hunk 23]
```diff
@@ -1,10 +1,49 @@
...
+question = 'Who was once considered the best kick boxer in the world, however he has been involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring?'
+
+llm = ChatOpenAI(openai_api_key="YOUR_API_KEY")
+agent = ReActAgent(llm=llm)
+out = agent.generate(question=question)
+
+```
+
+
+Here's what the output would look like!```
```
Comment (alckasoc) [inline README.md:31]:
```suggestion
Here's what the output would look like!
```

File: README.md [hunk 24]
```diff
@@ -1,10 +1,49 @@
...
+```
+Thought : I need to search for the best kick boxer in the world, and then find any controversies or crimes they have been involved in.
+Action : Search[best kick boxer in the world]
+Observation : (Result 1/1) Ramon Dekkers is considered by many to be the best kickboxer in the world.
+Thought : It mentions "unsportsmanlike conducts" and crimes of violence. I need to find more information about Ramon Dekkers.
+Action : Lookup[crimes]
+Observation: (Result 1/1) Dekkers was involved in a number of controversies relating to his "unsportsmanlike conducts" in the sport and crimes of violence outside of the ring.
+Thought: Ramon Dekkers has been involved in controversies and crimes. I need to find more information about them.
+...
+
```
Comment (alckasoc) [inline README.md:41]:
```suggestion
```

**Review by alckasoc — APPROVED** [review]

**Pull request merged**